Azure/azure-sdk-for-js
# [identity] Latest API updates for Azure pipelines credential

### Packages impacted by this PR
@azure/identity

### Issues associated with this PR


### Describe the problem that is addressed by this PR

API changes for the `AzurePipelinesCredential`
- Update OIDC API version to 7.1 instead of 7.1-preview.1
-  const oidcRequestUrl = `${process.env.SYSTEM_OIDCREQUESTURI}?api-version=${OIDC_API_VERSION}&serviceConnectionId=${serviceConnectionId}`;
- Update the error messages for system variables check to verify existence of `SYSTEM_OIDCREQUESTURI` and `SYSTEM_ACCESSTOKEN`
- Update error messages to append the response body to the AuthenticationError in case of oidc token was not detected in the response. Append the response body in the error message only if the response status is not 200.
- API change: 
- 1. Make the clientId, tenantId,  serviceConnectionId  and system access token a required parameter

Testing:
1. Ensure current test with the following values from the test keyvault works:AZURE_SERVICE_CONNECTION_CLIENT_ID,AZURE_SERVICE_CONNECTION_TENANT_ID,AZURE_SERVICE_CONNECTION_ID
2. Add a test that runs inside the AzurePowerShell@5 task or AzureCLI@2 task that takes in only the systemAccessToken and rest of the 3 values from the env vars - AZURESUBSCRIPTION_CLIENT_ID, AZURESUBSCRIPTION_TENANT_ID, AZURESUBSCRIPTION_SERVICE_CONNECTION_ID,

### Are there test cases added in this PR? _(If not, why?)_
- Yes

### Provide a list of related PRs _(if any)_


### Command used to generate this PR:**_(Applicable only to SDK release request PRs)_

### Checklists
- [ ] Added impacted package name to the issue description
- [ ] Does this PR needs any fixes in the SDK Generator?** _(If so, create an Issue in the [Autorest/typescript](https://github.com/Azure/autorest.typescript) repository and link it here)_
- [ ] Added a changelog (if necessary)
